lib/user_data_parser/data_set/data_set.rb

```diff
@@ -0,0 +1,13 @@
+module UserDataParser
+  class DataSet
+    attr_reader :rows
+
+    def initialize(rows = [])
+      @rows = rows
+    end
+
+    def add_rows(rows)
+      DataSet.new(@rows + rows)
+    end
+  end
+end
```

Comment:
See my comment below on ```OutputHandler```. I think a similar line of thinking may apply here. This looks like it might just be a fancy wrapper for an array.